### PR TITLE
Integrate donation with /api/donations/ service

### PR DESCRIPTION
**Pull request by gyokkoo**
- Persist the current logged in user in Local Storage
- Perform GET and POST request on /api/donations/ endpoint to get all and update donation data

### CODE REVIEW

File: src/components/Donate/DonateDialog.js [hunk 1]
```diff
@@ -4,47 +4,83 @@ import "@ui5/webcomponents/dist/Dialog";
...
+      const url = BLOOD_BANK_API + '/donations/';
+      fetch(url, {
+        method: 'POST',
+        body: JSON.stringify({
+          "userID": localStorage.getItem("userID") || "",
+          "bloodcenter": hospital,
+          "status": "In Progress"
+        })
+      }).then(() => {
+        console.log("Your donation request sent successfully! Please refresh to see the latest status")
```
Comment (m-ignatov) [inline src/components/Donate/DonateDialog.js]: Maybe refresh all data when created?

Also, 'date' field seems to be empty when creating new donation request.

File: src/components/Donate/DonateDialog.js [hunk 2]
```diff
@@ -4,47 +4,83 @@ import "@ui5/webcomponents/dist/Dialog";
...
+    this.onComboboxChange = this.onComboboxChange.bind(this);
   }
 
   componentDidMount() {
     if (this.dialogRef.current) {
       this.createButtonRef.current.addEventListener("click", this.create);
       this.cancelButtonRef.current.addEventListener("click", this.close);
     }
+
+    this.comboboxRef.current.addEventListener('change', this.onComboboxChange);
```
Comment (m-ignatov) [inline src/components/Donate/DonateDialog.js]: Add check:
if (this.comboboxRef.current) {....}

File: src/components/Donate/Donate.js [hunk 3]
```diff
@@ -19,41 +20,31 @@ class Donate extends Component {
...
   }
 
-  componentDidMount() {
+  async componentDidMount() {
+    const url = BLOOD_BANK_API + '/donations/';
+    const response = await fetch(url, {
+      method: 'GET',
+    });
+    const data = await response.json();
+    console.debug(data);
```
Comment (m-ignatov) [inline src/components/Donate/Donate.js]: debug

File: src/components/Donate/Donate.css [hunk 4]
```diff
@@ -7,6 +7,11 @@
   margin: 2rem 0 1rem 0;
 }
 
+.loading-container {
+  text-align: center;
+  display: block;
```
Comment (m-ignatov) [inline src/components/Donate/Donate.css:11]: Maybe vertically center also?

File: src/components/Donate/Donate.js [hunk 5]
```diff
@@ -93,10 +84,18 @@ class Donate extends Component {
...
 
     let filteredDonations = this.filterDonations(donations);
 
+    if (loading) {
+      return (
+        <div className="loading-container">
+          <ui5-busyindicator active size="Medium"></ui5-busyindicator>
+        </div>
+      )
+    }
```
Comment (m-ignatov) [inline src/components/Donate/Donate.js]: You can move the busy indicator as a parent of the `<ui5-table>`

```
<ui5-busyindicator>
    <ui5-table>... </ui5-table>
</ui5-busyindicator>
```

Comment (gyokkoo) [reply src/components/Donate/Donate.js]: Fixed

File: src/components/Donate/Donate.css (hunk 4 above)
Comment (gyokkoo) [reply src/components/Donate/Donate.css:11]: Fixed

File: src/components/Donate/Donate.js (hunk 3 above)
Comment (gyokkoo) [reply src/components/Donate/Donate.js]: Changed it to console.log for consistency

File: src/components/Donate/DonateDialog.js (hunk 2 above)
Comment (gyokkoo) [reply src/components/Donate/DonateDialog.js]: Fixed

File: src/components/Donate/DonateDialog.js (hunk 1 above)
Comment (gyokkoo) [reply src/components/Donate/DonateDialog.js]: Fixed but I am not sure if it is the 'React' way... please verify.

**Review by m-ignatov — APPROVED** [review]

**Comment by m-ignatov** [comment]
Да влиза!

**Pull request merged**